test/suites/base.bash

```diff
@@ -16,6 +16,24 @@ base_tests() {
     expect_stat 'files in cache' 1
     expect_equal_object_files reference_test1.o test1.o
 
+    CCACHE_COMPILER=$COMPILER $CCACHE non_existing_compiler_will_be_overriden_anyway -c test1.c
```

Comment:
Thanks for the new test cases!

I suggest moving them to a separate "CCACHE_COMPILER" test (see the other `TEST "CCACHE_..."` tests in the same file) since I think that using CCACHE_COMPILER doesn't quality as a base case.